lib/widgets/content.dart

```diff
@@ -200,8 +201,11 @@ class MessageImage extends StatelessWidget {
                 alignment: Alignment.center,
                 color: const Color.fromRGBO(0, 0, 0, 0.03),
                 child: Image.network(
-                  adjustedSrc,
+                  resolvedSrc,
                   filterQuality: FilterQuality.medium,
+                  headers: isUrlOnRealm(resolvedSrc, store.account)
+                    ? Map.fromEntries([authHeader(store.account)])
+                    : null,
```

Comment:
This pattern repeats a few times, and it's security-sensitive to get right: the URL needs to be checked to be on the account's realm before we go and use the account's API key in the header we'll send when fetching it.

So it'd be helpful to encapsulate it a bit more, to look something like:
```dart
  headers: maybeAuthHeader(resolvedSrc, store.account),
```

Better yet, we could encapsulate the call to the `Image.network` constructor — it's important that the URL we check here is the same one we use these headers to go and fetch, and that would let us put that connection in one place.

To do that, we can just make a function that takes all the same parameters as `Image.network` except `headers`, plus takes an `Auth`, and it passes all the rest through to `Image.network`. So this would look like:
```dart
  child: zulipNetworkImage(
    resolvedSrc,
    auth: store.account,
    filterQuality: FilterQuality.medium)
```

Reply:
What do you think of a new stateless widget that grabs the `Auth` by itself, directly from the established channel `PerAccountStoreWidget.of(context)`? Then I think it's easier to track the `Auth` from origin to destination, since it takes fewer steps.

Reply:
Sure, that sounds good.

Reply:
I got to wondering whether we could easily add a test for that new stateless widget…I think the next step in that direction is this bit in 882c8aed328acd59fe3ca1dacad90fe545c4ca36:

>     (Widgets still hardcode the "live" GlobalStore, so we'll need
>     to make some further changes when we go to write tests for those
>     widgets.  But this is already enough to write tests for any
>     logic we put on GlobalStore.)

since in the test setup we'll want the new widget's `PerAccountStoreWidget.of(context)` call to work (and in particular to work without contacting an actual Zulip server 😜).

So maybe not something to do right this moment, but worth a TODO? 🙂